extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -387,17 +398,48 @@ export class AmpLiveList extends AMP.BaseElement {
         this.itemsSlot_.appendChild(orphan);
       } else {
         const orphanSortTime = this.getSortTime_(orphan);
-        for (let child = this.itemsSlot_.firstElementChild; child;
-          child = child.nextElementSibling) {
-          const childSortTime = this.getSortTime_(child);
-          if (orphanSortTime >= childSortTime) {
-            this.itemsSlot_.insertBefore(orphan, child);
-            count++;
-            break;
-          // We've exhausted the children list and the current orphan
-          // can be the last item.
-          } else if (!child.nextElementSibling) {
-            this.itemsSlot_.appendChild(orphan);
+
+        if (this.isReverseOrder_) {
+          for (let child = this.itemsSlot_.lastElementChild; child;
+            child = child.previousElementSibling) {
+            const childSortTime = this.getSortTime_(child);
+            if (orphanSortTime >= childSortTime) {
+              if (child.nextElementSibling) {
```

Comment:
`insertAfterOrAtStart` from `dom.js` would do this.

Reply:
its 90% of what I need but I need the logic of "keep retrying until there is no more previousElementSibling" and breaking early